keras_nlp/models/gpt2/gpt2_preprocessor.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""GPT2 preprocessor layer."""
+
+import copy
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.models.gpt2.gpt2_presets import backbone_presets
+from keras_nlp.models.gpt2.gpt2_tokenizer import GPT2Tokenizer
+from keras_nlp.models.preprocessor import Preprocessor
+from keras_nlp.utils.keras_utils import (
+    convert_inputs_to_list_of_tensor_segments,
+)
+from keras_nlp.utils.keras_utils import pack_x_y_sample_weight
+from keras_nlp.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class GPT2Preprocessor(Preprocessor):
```

Comment:
What is this preprocessor used for vs the casual LM one? I guess if you want to dense output from the backbone?

Somehow my mental model for GPT-2 is seq2seq (or prompt2seq?)

Reply:
GPT2 backbone is still an "encoder" (although we call that decoder since it uses causal mask), quite similar to BERT.

Re the preprocessor, Causal LM preprocessor is built on top of `GPT2Preprocessor`, and adds the functionality to generate the `y` and `sample_weight`.